algorithms/src/Kokkos_Random.hpp

```diff
@@ -668,6 +668,23 @@ struct Random_UniqueIndex<Kokkos::Experimental::HIP> {
 };
 #endif
 
+#ifdef KOKKOS_ENABLE_SYCL
+template <>
+struct Random_UniqueIndex<Kokkos::Experimental::SYCL> {
+  using locks_view_type = View<int*, Kokkos::Experimental::SYCL>;
+  KOKKOS_FUNCTION
+  static int get_state_idx(const locks_view_type& locks_) {
+    // FIXME_SYCL get an index that is specific to each thread.
```

Comment:
Expand that comment that it is not about correctness but about starting from a different index to reduce contention